File: app/src/main/java/com/system/m4/views/transaction/TransactionManagerContract.java
package com.system.m4.views.transaction;

import com.system.m4.views.vos.PaymentTypeVO;
import com.system.m4.views.vos.TagVO;
import com.system.m4.views.vos.Transaction;
import com.system.m4.views.vos.VOInterface;

import java.util.Date;

/**
 * Created by Enir on 21/04/2017.
 * Contrato para TransactionManager
 */

interface TransactionManagerContract {

    interface View {

        void configureModel(Transaction transaction);

        void setPaymentDate(String value);

        void setPurchaseDate(String value);

        void setValue(String value);

        void setTags(String value);

        void setPaymentType(String value);

        void setContent(String value);

        void showValueDialog(Double value);

        void showContentDialog(String value);

        void showPaymentDateDialog(Date date);

        void showPurchaseDateDialog(Date date);

        void dismissDialog(VOInterface vo);

        void showError(String error);

        void showSuccessMessage(int template, int param);

        void showError(int template, int param);
    }

    interface Presenter {

        void setPaymentDate(Date date);

        void setPurchaseDate(Date date);

        void setValue(Double value);

        void setTags(TagVO tagVO);

        void setPaymentType(PaymentTypeVO paymentTypeVO);

        void setContent(String content);

        void requestValueDialog(String text);

        void requestContentDialog(String text);

        void requestPaymentDateDialog(String text);

        void requestPurchaseDateDialog(String text);

        void clearContent();

        void clearPaymentType();

        void clearPrice();

        void clearPurchaseDate();

        void clearPaymentDate();

        void save();

        void init(Transaction transaction);
    }
}

File: app/src/main/java/com/system/m4/kotlin/transaction/TransactionManagerContract.kt
package com.system.m4.kotlin.transaction

import android.content.Context
import com.system.m4.views.vos.PaymentTypeVO
import com.system.m4.views.vos.TagVO
import com.system.m4.views.vos.TransactionVO
import com.system.m4.views.vos.VOInterface
import java.util.*

/**
 * Created by Enir on 21/04/2017.
 * Contrato para TransactionManager
 */

interface TransactionManagerContract {

    interface View {

        fun setToolbarTitle(titleString: String)
        fun setPaymentDate(value: String)
        fun setPurchaseDate(value: String)
        fun setPrice(value: String)
        fun setRefund(value: String)
        fun setTags(value: String)
        fun setPaymentType(value: String)
        fun setContent(value: String)
        fun setParcels(value: String)
        fun setAlreadyPaid(value: String)

        fun showPriceDialog(value: Double?)
        fun showRefundDialog(value: Double?)
        fun showContentDialog(value: String?)
        fun showParcelsDialog(value: String?)
        fun showPaymentDateDialog(date: Date)

        fun showPurchaseDateDialog(date: Date)
        fun dismissDialog(vo: VOInterface<*>)
        fun showSuccessMessage(template: Int, param: Int)
        fun showError(error: String)
        fun showError(template: Int, param: Int)
        fun getContext(): Context
    }

    interface Presenter {

        /*
         * INIT
         */

        fun init(transaction: TransactionVO)

        /*
         * ON CLICK EVENTS
         */

        fun onPurchaseDateClick(text: String)
        fun onPaymentDateClick(text: String)
        fun onPriceClick(text: String)
        fun onRefundClick(text: String)
        fun onContentClick(text: String)
        fun onParcelsClick(text: String)
        fun onAlreadyPaidClick(value: String)

        /*
         * ON CLICK EVENTS
         */

        fun onPaymentTypeLongClick(): Boolean
        fun onPurchaseDateLongClick(): Boolean
        fun onPaymentDateLongClick(): Boolean
        fun onPriceLongClick(): Boolean
        fun onRefundLongClick(): Boolean
        fun onContentLongClick(): Boolean
        fun onParcelsLongClick(): Boolean

        /*
         * SETTERS
         */

        fun setTags(tagVO: TagVO)
        fun setPaymentType(paymentTypeVO: PaymentTypeVO?)
        fun setPurchaseDate(date: Date?)
        fun setPaymentDate(date: Date?)
        fun setPrice(value: Double?)
        fun setRefund(value: Double?)
        fun setContent(content: String?)
        fun setParcels(parcels: String?)
        fun setAlreadyPaid(alreadyPaid: Boolean)

        fun save()
    }
}